lib/collection/src/shards/shard_holder/mod.rs

```diff
@@ -207,6 +207,35 @@ impl ShardHolder {
                 .add(*shard_id);
         }
 
+        // Restore resharding hash ring if resharding is active and haven't reached WriteHashRingCommitted yet
```

Comment:
Above, we construct new hash rings as if no resharding is happening.

When resharding, we expect them to reach a different state though. That's done below.

When resharding up, we should have a resharding hash ring. The old ring should not include the new resharding shard.

When resharding down, we should have a resharding hash ring. The new hash ring should not include the shard that is about to be removed.